src/Core/Domain/ImageSettings/CommandHandler/RegenerateThumbnailsHandler.php

```diff
@@ -54,59 +57,83 @@ public function __construct(
      * {@inheritdoc}
      *
      * @throws ImageTypeException
+     * @throws RegenerateThumbnailsException
      */
     public function handle(RegenerateThumbnailsCommand $command): void
     {
         // Get all languages
         $languages = $this->langRepository->findAll();
 
-        // Set images to regenerate with all theirs specific directories
-        $process = [
-            ['type' => 'categories', 'dir' => _PS_CAT_IMG_DIR_],
-            ['type' => 'manufacturers', 'dir' => _PS_MANU_IMG_DIR_],
-            ['type' => 'suppliers', 'dir' => _PS_SUPP_IMG_DIR_],
-            ['type' => 'products', 'dir' => _PS_PRODUCT_IMG_DIR_],
-            ['type' => 'stores', 'dir' => _PS_STORE_IMG_DIR_],
-        ];
-
-        // Launching generation process
-        foreach ($process as $proc) {
+        // Iterate all images categories with thumbnails
+        foreach (ImageDomain::getDomainsWithThumbnails() as $imageCategory) {
             // Check if this kind of image is selected, if not, skip
-            if ($command->getImage() !== 'all' && $command->getImage() !== $proc['type']) {
+            if ($command->getImage() !== ImageDomain::ALL && $command->getImage() !== $imageCategory) {
                 continue;
             }
 
             // Getting formats generation (all if 'all' selected)
-            if ($command->getImageTypeId() === 0) {
-                $formats = $this->imageTypeRepository->findBy([$proc['type'] => 1]);
-            } else {
-                $formats = $this->imageTypeRepository->findBy(['id' => $command->getImageTypeId(), $proc['type'] => 1]);
+            $imageTypesCriteria = [$imageCategory->value => 1];
+            if ($command->getImageTypeId() instanceof ImageTypeId) {
+                $imageTypesCriteria['id'] = $command->getImageTypeId()->getValue();
```

Comment:
To make it equivalent, I believe you'd need to do

```suggestion
                $imageTypesCriteria = ['id' => $command->getImageTypeId()->getValue()];
```
The condition is replaced, not added

Reply:
resolved by the previous comment response